server/src/main/java/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -2607,13 +2608,13 @@ protected void connectAndRestartAgentOnHost(HostVO host, String username, String
         final com.trilead.ssh2.Connection connection = SSHCmdHelper.acquireAuthorizedConnection(
                 host.getPrivateIpAddress(), 22, username, password);
         if (connection == null) {
-            throw new CloudRuntimeException("SSH to agent is enabled, but failed to connect to host: " + host.getPrivateIpAddress());
+            throw new CloudRuntimeException(String.format("SSH to agent is enabled, but failed to connect to %s via IP address %s.", host, host.getPrivateIpAddress()));
```

Comment:
I would rather use `[]` as var delimiter in some cases; IMHO, it is easier to recognize what is a variable and what is fixed in the final message, e.g.:   
`SSH to agent is enabled, but failed to connect to %s via IP address [%s].`.